infra/RepoManager.py

```diff
@@ -0,0 +1,202 @@
+
+
+class RepoManagerError(Exception):
+  """Class to describe the exceptions in RepoManager."""
+
+  def __init__(self, message):
+    """ Init the exception class.
+
+    Args:
+      message: the message to propagate to user
+    """
+    super().__init__(message)
+
+
+class RepoManager(object):
+  """Class to manage git repos from python.
+
+  Attributes:
+    repo_url: The location of the git repo
+    base_dir: The location of where the repo clone is stored locally
+    repo_name: The name of the github project
+    repo_dir: The location of the main repo
+  """
+
+  def __init__(self, repo_url, base_dir):
+    """Constructs a repo manager class.
+
+    Args:
+      repo_url: The github url needed to clone
+      base_dir: The full filepath where the git repo is located
+    """
+
+    self.repo_url = repo_url
+    self.base_dir = base_dir
+    self.repo_name = self.repo_url.split('/')[-1].strip('.git')
+    self.repo_dir = os.path.join(self.base_dir, self.repo_name)
+    self._clone()
+
+  def _clone(self):
+    """Creates a clone of the repo in the specified directory.
+
+      Raises:
+        RepoManagerError if the repo was not able to be cloned
+    """
+    if not os.path.exists(self.base_dir):
+      os.makedirs(self.base_dir)
+    self.remove_repo()
+    _, err = self._run_command(['git', 'clone', self.repo_url],
+                               self.base_dir,
+                               check_result=True)
+    if not self._is_git_repo():
+      raise RepoManagerError('%s is not a git repo' % self.repo_url)
+
+  def _run_command(self, command, location='.', check_result=False):
+    """ Runs a shell command in the specified directory location.
+
+    Args:
+      command: The command as a list to be run
+      location: The directory the command is run in
+      check_result: Should an exception be thrown on failed command
+
+    Returns:
+      The stdout of the command, the stderr of the command
+
+    Raises:
+      RepoManagerError when running a command resulted in an error
+    """
+    process = subprocess.Popen(command, stdout=subprocess.PIPE, cwd=location)
+    out, err = process.communicate()
+    if err is not None and check_result:
+      raise RepoManagerError('Error: %s running command: %s' % (err, command))
+    if out is not None:
+      out = out.decode('ascii')
+    return out, err
+
+  def _is_git_repo(self):
+    """Test if the current repo dir is a git repo or not.
+
+    Returns:
+      True if the current repo_dir is a valid git repo
+    """
+    git_path = os.path.join(self.repo_dir, '.git')
+    return os.path.isdir(git_path)
+
+  def commit_exists(self, commit):
+    """Checks to see if a commit exists in the project repo.
+
+    Args:
+      commit: The commit SHA you are checking 
+
+    Returns:
+      True if the commit exits in the project
+    """
+
+    # Handle the exception case, if empty string is passed _run_command will
+    # return true
+    if commit.rstrip() == '':
+      return False
```

Comment:
should we raise an exception with ValueError here instead? An empty commit string doesn't seem to make much sense.